src/index/ivf_raft/ivf_raft.cuh

```diff
@@ -37,6 +37,58 @@
 
 namespace knowhere {
 
+namespace raft_res_pool {
+
+struct context {
+    static constexpr std::size_t default_size{16};
```

Comment:
Is this intending to create 16 `rat::device_resources` instances up front in the same thread? All the `device_resources` instances are going to be using the same `cudaStream_t` underneath so it's not going to be asynchronous or otherwise parallelizable.

Reply:
no, we create a group of stream, std::vector<rmm::cuda_stream> stream_;